src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.String.cs

```diff
@@ -34,8 +34,35 @@ public static string Serialize<TValue>(TValue value, JsonSerializerOptions? opti
         /// </remarks>
         public static string Serialize(object? value, Type inputType, JsonSerializerOptions? options = null)
         {
-            VerifyValueAndType(value, inputType);
-            return WriteCoreString(value, inputType, options);
+            if (inputType == null)
+            {
+                throw new ArgumentNullException(nameof(inputType));
+            }
+
+            if (value != null && !inputType.IsAssignableFrom(value.GetType()))
+            {
+                ThrowHelper.ThrowArgumentException_DeserializeWrongType(inputType, value);
+            }
+
+            return Serialize<object>(value!, inputType, options);
```

Comment:
I prefer the explict code here since the very next method in the same file has the same name+arguments except the first arg is `object?` instead of `TValue`